### PR TITLE
[4.0] Move inline onclick handler to js script

### PR DESCRIPTION
**Pull request by laoneo**
Moves the onclick code to the js script in finder.

Ping @dgt41.

### CODE REVIEW

File: media/com_finder/js/finder-edit.js [hunk 1]
```diff
@@ -10,6 +10,18 @@ Joomla = window.Joomla || {};
 
 	document.addEventListener('DOMContentLoaded', function() {
 
+		var buttons = document.querySelectorAll('.toggle-all');
```
Comment (dgrammatiko) [inline media/com_finder/js/finder-edit.js]: Make this: `var buttons = [].slice.call(document.querySelectorAll('.toggle-all'));` to make the list of nodes an array

File: media/com_finder/js/finder-edit.js [hunk 2]
```diff
@@ -10,6 +10,18 @@ Joomla = window.Joomla || {};
...
+
+		for (var i = 0; i < buttons.length; i++) {
```
Comment (dgrammatiko) [inline media/com_finder/js/finder-edit.js]: And then replace the for by a forEach:
```js
buttons.foreEach(function(button) {
	button.addEventListener('click', function(e) {
 		var filterNodes = [].slice.call(document.querySelectorAll('.filter-node'));
 
 		filterNodes.forEach(function(node) {
 			node.click();
 		});
});
```

**Review by dgrammatiko — APPROVED** [review]

File: media/com_finder/js/finder-edit.js (hunk 1 above)
Comment (laoneo) [reply media/com_finder/js/finder-edit.js]: Why that, you can iterate also over the node list?

**Comment by laoneo** [comment]
I leave them with a for loop ;-=

**Comment by laoneo** [comment]
Changed to forEach. Ready to merge.

**Review by C-Lodder — APPROVED** [review]

**Comment by dneukirchen** [comment]
1) Its not safe to bind behavior to css classes. It will lead to conflicts with templates, modules or other scrips. Especially when they are commonly used like `.toggle-all`. 

2) Why are we triggering the click event on the checkbox instead of just checking it?

3) Its a common task to check/uncheck checkboxes in joomla. We should have a global helper for it (and we already have one which is tightly coupled to the dom and not flexible enough. Yes, im looking at you Joomla.checkAll())

**Comment by laoneo** [comment]
We are talking here only about a simple script in the finder component.

**Pull request closed without merging**